harness/determined/pytorch/_trainer.py

```diff
@@ -176,11 +170,39 @@ def fit(
 
             smaller_is_better = bool(self._info.trial._config["searcher"]["smaller_is_better"])
             searcher_metric_name = self._info.trial._config["searcher"]["metric"]
+
             steps_completed = int(self._info.trial._steps_completed)
             global_batch_size = self._info.trial.hparams.get("global_batch_size", None)
             if global_batch_size:
                 global_batch_size = int(global_batch_size)
 
+            # Backwards compatibility: try to parse legacy `searcher.max_length` if `max_length`
+            # isn't passed in.
+            if max_length is None:
+                max_length_val = self._core.searcher._parse_searcher_max_length(
+                    self._info.trial._config
+                )
+                if max_length_val:
+                    logger.warning(
+                        "Configuring `max_length` from the `searcher.max_length` experiment config, "
+                        "which is deprecated and will be removed in a future release. Please "
```

Comment:
Please indicate which version it was deprecated in.  I've been using the `XXYYZZ` placeholder for the version, which I'll go through and update before we land it.